Thecarisma-syntax.md

```diff
@@ -0,0 +1,620 @@
+## SOURCE FILE
+- Source file can be empty, otherwise it must contain a valid set of UTF-8 codepoints.
+
+- Support for UTF-8 identifiers is planned.
+
+
+## COMMENTS
+#### Single-line comments
+- Single-line comments start with `#` and `//`
+    ```python
+    # Hello, World!
+    ```
+    ```java
+    // Hello, World!
+    ```
+
+    The singline comment starts with `#` so the plain source can be executed natively using shebang `(#!)` or system command can be executed at the start of the source file.
+
+#### Multiline comments
+- Nested multiline comment is allowed.
+    ```rust
+    /*
+     * Multiline comments can span multiple lines.
+     * They begin with `/*`.
+     * And they end with `*/`.
+     */
+    ```
+
+## SUBJECT
+- Subject consist of a subject name and value
+
+    ```cpp
+    const name = "John"
+    ```
+
+    Here `name` is the name of the subject and `"John"` the value or object.
+
+- The `const` keyword is used for subject which value cannot change.
+    ```cpp
+    const name = "John"
+    name = "Doe" // Invalid!
+    ```
```

Comment:
I don't agree with using const for constants because it is verbose and would be tiresome to keep typing since it is going to be used a lot code. let as a constant is not a new syntax on the block. They have been popularised by Swift and Rust. We could also use val as popularised by Kotlin and Scala, but I think let is more expressive. 

In modern languages, constants tend to be used quite a lot, because there is these notion of immutability by default, therefore these languages try to keep the keyword as short as possible. Usually as three letter words. val and `let`being the notable ones.

If you do a quick grep on any repo with code written in modern languages, you will notice constants are declared way more than variables.